Kernel/Arch/x86/i386/Processor.cpp

```diff
@@ -75,43 +76,40 @@ FlatPtr Processor::init_context(Thread& thread, bool leave_crit)
         m_in_critical = 1; // leave it without triggering anything or restoring flags
     }
 
-    u32 kernel_stack_top = thread.kernel_stack_top();
+    // TODO: handle NT?
+    VERIFY((cpu_flags() & 0x24000) == 0); // Assume !(NT | VM)
 
-    // Add a random offset between 0-256 (16-byte aligned)
-    kernel_stack_top -= round_up_to_power_of_two(get_fast_random<u8>(), 16);
+    // Get stack top and add a random offset between 0-256 (16-byte aligned)
+    FlatPtr const kernel_stack_top = thread.kernel_stack_top() - round_up_to_power_of_two(get_fast_random<u8>(), 16);
 
-    u32 stack_top = kernel_stack_top;
+    StackWriter stack_writer(kernel_stack_top);
 
-    // TODO: handle NT?
-    VERIFY((cpu_flags() & 0x24000) == 0); // Assume !(NT | VM)
+    // we want to end up 16-byte aligned, %esp + 4 should be aligned
+    constexpr u32 padding_value { 0 };
```

Comment:
Please pick a construction style and keep it consistent, either `Foo foo(bar)` or `Foo foo { bar }`.

Reply:
From what I've seen in other parts of the source code, there is no clear preference, so I decided to use parentheses for traditional constructors and braces for fundamental types so as to get the benefit of narrowing conversion protection.

I will change everything to use braces if you're adamant about it, however. What do you think?

Reply:
We don't prefer one over the other, but we usually prefer keeping the ones next to each other consistent.